lambda_code/tests/test_immunization_pre_validator_model.py

```diff
@@ -514,3 +522,210 @@ def test_model_pre_validate_reason_code_coding_displays(self):
+        """
+        ValidatorModelTests.test_string_value(
+            self,
+            field_location="contained[?(@.resourceType=='QuestionnaireResponse')]"
+            + ".item[?(@.linkId=='UserId')].answer[0].valueCoding.code",
+            valid_strings_to_test=["LESTER_TESTER-1234"],
+        )
+
+    def test_model_pre_validate_user_name_code(self):
+        """
+        Test pre_validate_user_name_code accepts valid values and rejects invalid values
+        """
+        ValidatorModelTests.test_string_value(
+            self,
+            field_location="contained[?(@.resourceType=='QuestionnaireResponse')]"
+            + ".item[?(@.linkId=='UserName')].answer[0].valueCoding.code",
+            valid_strings_to_test=["LESTER TESTER"],
+        )
+
+    def test_model_pre_validate_user_email_code(self):
+        """
+        Test pre_validate_user_email_code accepts valid values and rejects invalid values
+        """
+        ValidatorModelTests.test_string_value(
+            self,
+            field_location="contained[?(@.resourceType=='QuestionnaireResponse')]"
+            + ".item[?(@.linkId=='UserEmail')].answer[0].valueCoding.code",
+            valid_strings_to_test=["lester.tester@test.com"],
+        )
+
+    def test_model_pre_validate_submitted_time_stamp_code(self):
+        """
+        Test pre_validate_submitted_time_stamp_code accepts valid values and rejects invalid values
+        """
+        ValidatorModelTests.test_date_time_value(
+            self,
+            field_location="contained[?(@.resourceType=='QuestionnaireResponse')]"
+            + ".item[?(@.linkId=='SubmittedTimeStamp')].answer[0].valueCoding.code",
+        )
+
+    def test_model_pre_validate_location_identifier_value(self):
+        """
+        Test pre_validate_location_identifier_value accepts valid values and rejects invalid values
+        """
+        ValidatorModelTests.test_string_value(
+            self,
+            field_location="location.identifier.value",
+            valid_strings_to_test=["B0C4P", "140565"],
+        )
+
+    def test_model_pre_validate_location_identifier_system(self):
+        """
+        Test pre_validate_location_identifier_system accepts valid values and rejects invalid values
+        """
+        ValidatorModelTests.test_string_value(
+            self,
+            field_location="location.identifier.system",
+            valid_strings_to_test=["https://fhir.hl7.org.uk/Id/140565"],
+        )
+
+    def test_model_pre_validate_reduce_validation_code(self):
+        """
+        Test pre_validate_reduce_validation_code accepts valid values and rejects invalid values
+        """
+        ValidatorModelTests.test_string_value(
+            self,
+            field_location="contained[?(@.resourceType=='QuestionnaireResponse')]"
+            + ".item[?(@.linkId=='ReduceValidation')].answer[0].valueCoding.code",
+            valid_strings_to_test=["True", "False"],
```

Comment:
Needs to test for the enums rule (use arguments predefined_values = ("True", "False") and invalid_strings_to_test=....[insert list of some invalid non-empty strings here]).

Reply:
done